code/modules/hydroponics/plant_genes.dm

```diff
@@ -464,11 +472,23 @@
 	if(!L.reagents && !L.can_inject(null, 0))
 		return FALSE
 
-	var/injecting_amount = max(1, G.seed.potency*0.2) // Minimum of 1, max of 20
-	var/fraction = min(injecting_amount/G.reagents.total_volume, 1)
+	// Mechanism: It will always inject the amount as "injecting_amount" variable no matter the total size is
+	var/prick_efficiency = 0.02 // default (maximum: 2u)
+	if(locate(/datum/plant_gene/trait/squash) in G.seed.genes)
+		prick_efficiency = 0.01 // how can a smooth plant be deadly? (maximum: 1u)
+	else if(istype(G, /obj/item/seeds/nettle/death)) // as long as it has not Liquid Content
+		prick_efficiency = 0.06 // bonus to "death" nettle (maximum: 6u)
+	else if(G.force)
+		prick_efficiency = 0.04 // bonus to blunt plants (miaxmum: 4u)
+	if(!COOLDOWN_LIST_FINISHED(src, recent_victims, L)) // until 10s cooldown time from a victim is finished, the effective becomes half
+		prick_efficiency = round(prick_efficiency/2, 0.01) // i.e) 4u to 2u
```

Comment:
A maximum of 2 units is just ridiculously low to the point where it isn't doing anything. The cooldown on this one doesn't make a lot of sense either since on throw/slip isn't highly repeated events like attacking.

Reply:
Sadly it should be so. It's already horrible source of force-injection of toxins. 6u even does powerful enough.